frontend/src/metabase-lib/Question.ts

```diff
@@ -500,7 +499,7 @@ class Question {
     const hasSinglePk =
       table?.fields?.filter(field => field.isPK())?.length === 1;
 
-    return this.isStructured() && !legacyQuery.hasAnyClauses() && hasSinglePk;
+    return this.isStructured() && Lib.isStageEmpty(query, -1) && hasSinglePk;
```

Comment:
Are we OK with this breaking if the last stage is empty but the preceding one is not?

Reply:
That's how it worked before.
`hasAnyClauses` takes only the last stage into account.

Reply:
Do legacy queries have stages as such?

Reply:
> Do legacy queries have stages as such?

Yeah, somewhat - multiple `StructuredQuery` instances are connected to each other in a tree-like fashion via `sourceQuery()`. Every instance represents a stage.

Checking only the last stage was actually @ranquild's suggestion: https://github.com/metabase/metabase/pull/37570#discussion_r1450198781 + https://github.com/metabase/metabase/pull/37570#issuecomment-1888826099.

> Are we OK with this breaking if the last stage is empty but the preceding one is not?

There shouldn't be a case like that - there won't be a last stage that is empty, since this `supportsImplicitActions` method is used only outside of the GUI editor.